### ISSUE TITLE
DocumentNotFound Exception

### ISSUE THREAD

**Issue by AV25242** [couchbaselabs/node-ottoman#442]
In the 04/13 meeting with the customer it was determined that the following mutation operations are not throwing back
DocumentNotFound Exception 

```
findOneAndUpdate
removeById
replaceById
updateById
findOne
```

Can someone please verify to see what could be the reason ? 
If its Nodejs SDK not throwing back the exception we will need to know ASAP.

**Comment by gsi-ariam** [comment]
in the `couchnode\lib\errors.js` file of the sdk adding this change:

```js
class CouchbaseError extends Error {
  constructor(errtext, baseerr, context, ...args) {
    super(errtext, ...args);
    this.cause = baseerr;
    this.context = context;
    this.name = this.constructor.name; // 👈👈👈 adding this line the errors comes out with the corresponding name
    Error.captureStackTrace(this, CouchbaseError);
  }
}

exports.CouchbaseError = CouchbaseError;
```

The errors would be showing with their corresponding names, example:

```js
const CatSchema = new Schema({
  name: String,
  age: Number,
});

const Cat = model('Cat', CatSchema);

try {
  await Cat.updateById('id-no-found', {})
} catch (e) {
  const { name, message } = e;
  console.log(`Name: ${JSON.stringify(name)}`);
  console.log(`Message: ${JSON.stringify(message)}`);
}

```

// OUTPUT with update
```sh
Name: "DocumentNotFoundError"
Message: "document not found"
```

// OUTPUT without update
```sh
Name: "Error"
Message: "document not found" (editado) 
```

**Comment by AV25242** [comment]
@gsi-ariam  thanks much, I have reported this information in the Nodejs ticket and made @brett19  aware of this, he will evaluate it.

**Comment by AV25242** [comment]
related to #356

**Comment by AV25242** [comment]
@gsi-alejandro and @gsi-ariam  the underlying SDK issue
https://issues.couchbase.com/browse/JSCBC-855 
seems to be getting released today with SDK 3.2 Beta
please do the necessary changes on Ottoman side

**Comment by gsi-ariam** [comment]
@AV25242 according to Brett SDK currently only supports our internal builds.
